File: split/makemark.pas
procedure makemark;
var
  p: halfword;
begin
  p := scantoks(false, true);
  p := getnode(2);
  mem[p].hh.b0 := 4;
  mem[p].hh.b1 := 0;
  mem[p + 1].int := defref;
  mem[curlist.tailfield].hh.rh := p;
  curlist.tailfield := p;
end;

File: cpp/makemark.cpp
#include "makemark.h"
#include "scantoks.h"
#include "getnode.h"

void makemark(void)
{
	auto p = scantoks(false, true);
	p = getnode(2);
	type(p) = mark_node;
	subtype(p) = 0;
	mem[p+1].int_ = defref;
	tail_append(p);
}
